component/src/main/java/com/celements/blog/article/Article.java

```diff
@@ -445,4 +486,183 @@ public String toString() {
     return "Article [docRef=" + getDocumentReference() + "]";
   }
 
+  public String getExtractPlainTextEncoded(String lang, boolean isViewtypeFull, int maxNumChars) {
+    String articleExtract = getExtract(lang, isViewtypeFull, maxNumChars);
+    try {
+      String plainExtract = new PlainTextCommand().convertHtmlToPlainText(articleExtract);
+      return plainExtract.replaceAll("\"", "&quot;");
+    } catch (ConvertToPlainTextException ctpte) {
+      LOGGER.error("Failed to convert article extract to plain text for article {}", articleDocRef,
+          ctpte);
+    }
+    return articleExtract;
+  }
+
+  public List<ImageUrlDim> getArticleImagesBySizeAsc(String lang) {
+    List<ImageUrlDim> articleImages = extractImagesList(getFullArticle(lang));
+    if (articleImages.isEmpty()) {
+      articleImages = extractImagesList(getExtract(lang, false, getMaxNumChars()));
+    }
+    return articleImages;
+  }
+
+  List<ImageUrlDim> extractImagesList(String content) {
```

Comment:
**Separation of concerns**: This method contains too much complex logic and is too broad for a private method inside the Article class (which is already far too large). move it to a service or util class